### PR TITLE
UI2 Scan Modal

### PR DESCRIPTION
**Pull request by samholmes**
This is to be able to install expo react native modules.
The motivation is to install expo-image-picker for new UI2 Scan Modal.#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

### CODE REVIEW

File: src/components/modals/ScanModal.js [hunk 1]
```diff
@@ -33,94 +38,157 @@ type DispatchProps = {
...
+            <View
+              style={{
+                ...styles.peepholeSpace,
+                minHeight: peepholeSize
+              }}
+              onLayout={onLayoutPeepholeSpace}
+            />
+            <View style={styles.buttonsContainer}>
+              <TouchableHighlight onPress={onPressFlashIcon} underlayColor={theme.secondaryButton}>
+                <View style={styles.iconButtonContainer}>
```
Comment (swansontec) [inline src/components/modals/ScanModal.js:157]: You can skip the `<View>` and put the style right on `<TouchableHighlight ...>`

File: src/components/modals/ScanModal.js [hunk 2]
```diff
@@ -33,94 +38,157 @@ type DispatchProps = {
...
+              </TouchableHighlight>
+            </View>
+          </View>
+        </>
       )
     }
 
     return (
-      <View style={styles.cameraArea}>
+      <View style={{ flex: 1, justifyContent: 'center' }}>
```
Comment (swansontec) [inline src/components/modals/ScanModal.js:175]: Do we want a named style for this one?

File: src/components/modals/ScanModal.js [hunk 3]
```diff
@@ -33,94 +38,157 @@ type DispatchProps = {
...
-    return (
-      <AirshipModal bridge={bridge} onCancel={this.close} borderRadius={theme.rem(1)} backgroundColor={theme.tileBackground}>
-        <ModalTitle>{title}</ModalTitle>
-        {this.renderCameraArea()}
-        <ModalCloseArrow onPress={this.close} />
-      </AirshipModal>
-    )
-  }
+  return (
+    <AirshipModal bridge={bridge} margin={[airshipMarginTop, 0, 0, 0]} padding={[0, 0, 0, 0]} backgroundColor={theme.modal} onCancel={close} overflow="hidden">
```
Comment (swansontec) [inline src/components/modals/ScanModal.js:182]: You can use `margin={[airshipMarginTop, 0, 0]}` and `padding={0}`, and the other numbers will fill out, just like on the web.

File: src/theme/variables/edgeDark.js [hunk 4]
```diff
@@ -97,6 +97,9 @@ export const edgeDark: Theme = {
   backgroundGradientLeft: palette.edgeNavy,
   backgroundGradientRight: palette.darkBlue,
 
+  // Overlay
+  overlayBackground: palette.gray,
```
Comment (swansontec) [inline src/theme/variables/edgeDark.js:101]: What is this? Looking at the theme, without searching the code-base for usage examples, I would have no idea. Would "cameraOverlay" make more sense?

**Review by swansontec — CHANGES_REQUESTED** [review]
This looks pretty good overall, but there are a good number of details that need to be tightened up.

File: android/app/src/main/java/co/edgesecure/app/MainApplication.java [hunk 5]
```diff
@@ -1,5 +1,8 @@
 package co.edgesecure.app;
 
+// com.myapp should be your package name
+import co.edgesecure.generated.BasePackageList;
```
Comment (swansontec) [inline android/app/src/main/java/co/edgesecure/app/MainApplication.java]: `(cd android; ./formatJava.sh)`, to sort the imports.

File: ios/Podfile [hunk 6]
```diff
@@ -1,9 +1,11 @@
 require_relative '../node_modules/react-native/scripts/react_native_pods'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
+require_relative '../node_modules/react-native-unimodules/cocoapods.rb'
 
-platform :ios, '10.0'
+platform :ios, '11.0'
```
Comment (swansontec) [inline ios/Podfile:5]: Wait! This is the same issue we ran into with that other camera library. We can't bump to iOS 11 until we get clearance from RJ.

File: src/components/modals/ScanModal.js [hunk 7]
```diff
@@ -119,9 +118,9 @@ const Component = (props: Props) => {
     if (cameraPermission === RNPermissions.RESULTS.BLOCKED) {
       return (
         <>
-          <T style={styles.cameraPermissionDeniedText}>{s.strings.scan_camera_permission_denied}</T>
+          <ModalMessage style={styles.cameraPermissionDeniedText}>{s.strings.scan_camera_permission_denied}</ModalMessage>
```
Comment (swansontec) [inline src/components/modals/ScanModal.js:121]: I don't think `ModalMessage` even takes a `style` property, so remove that.

File: src/components/modals/ScanModal.js [hunk 8]
```diff
@@ -159,13 +158,13 @@ const Component = (props: Props) => {
               <TouchableHighlight onPress={handleFlash} underlayColor={theme.secondaryButton}>
                 <View style={styles.iconButtonContainer}>
                   <Ionicon style={styles.icon} name="flash-outline" size={theme.rem(1.5)} />
-                  <T style={styles.iconButtonText}>{s.strings.fragment_send_flash}</T>
+                  <ModalMessage style={styles.iconButtonText}>{s.strings.fragment_send_flash}</ModalMessage>
```
Comment (swansontec) [inline src/components/modals/ScanModal.js:155]: Also here, I don't think `ModalMessage` even takes a style prop.

File: src/components/modals/ScanModal.js [hunk 9]
```diff
@@ -149,17 +149,17 @@ const Component = (props: Props) => {
             </View>
             <View style={[styles.peepholeSpace, { minHeight: peepholeSize }]} onLayout={handleLayoutPeepholeSpace} />
             <View style={styles.buttonsContainer}>
-              <TouchableHighlight onPress={handleFlash} underlayColor={theme.secondaryButton}>
-                <View style={styles.iconButtonContainer}>
+              <TouchableHighlight style={styles.iconButtonContainer} onPress={handleFlash} underlayColor={theme.secondaryButton}>
+                <>
```
Comment (swansontec) [inline src/components/modals/ScanModal.js:153]: I don't think you need the fragment here - `<TouchableHighlight>` takes multiple children.

**Review by swansontec — CHANGES_REQUESTED** [review]
Found a few more minor JSX issues.

File: src/components/modals/ScanModal.js (hunk 9 above)
Comment (samholmes) [reply src/components/modals/ScanModal.js:153]: I get an error if I don't have the fragment. React native throws an error saying expecting children lnegth of one.

Comment (swansontec) [reply src/components/modals/ScanModal.js:166]: Ok, then we should leave it in.

File: src/components/modals/ScanModal.js (hunk 8 above)
Comment (swansontec) [reply src/components/modals/ScanModal.js]: Still an issue.

File: src/theme/variables/edgeDark.js (hunk 4 above)
Comment (swansontec) [reply src/theme/variables/edgeDark.js:101]: Still an issue. We need a better name.

File: src/components/modals/ScanModal.js (hunk 7 above)
Comment (swansontec) [reply src/components/modals/ScanModal.js:134]: ModalMessage doesn't take a `style` prop, so please delete this.

**Review by swansontec — CHANGES_REQUESTED** [review]
Two issues remain.

**Review by swansontec — APPROVED** [review]
Oh, nice cleanups!

**Pull request merged**